dotnet/Gherkin/Parser.cs

```diff
@@ -75,8 +75,20 @@ public enum RuleType
     [System.Runtime.CompilerServices.CompilerGeneratedAttribute()]
     public partial class Parser<T>
     {
+        public Parser()
+            : this(new AstBuilder<T>())
+        {
+        }
+
+        public Parser(AstBuilder<T> astBuilder)
+        {
+            this.AstBuilder = astBuilder;
+        }
+
         public bool StopAtFirstError { get; set;}
 
+        public AstBuilder<T> AstBuilder { get; }

```

Comment:
I think expressing/clarifying intent is important. By making something a constant at the declaration says clearly "this does not change", so when the language support saying that - then why not use it? I do not know C# idioms that well to say that it is better to change it to a `readonly` field.